docs/release-notes/release-2024-06-11.md

```diff
@@ -0,0 +1,75 @@
+---
+title: 0.43.0 June 2024
+---
+
+# Release Notes Version 0.43.0 - June 2024
+
+See TypeSpec Core 0.57 [release notes](https://typespec.io/docs/release-notes/release-2024-06-11)
```

Comment:
```suggestion
See TypeSpec Core 0.57 [release notes](https://typespec.io/docs/release-notes/release-2024-06-10)
```
We made the other one 2024-06-10, looks like, so this link will need to be updated, or the other doc name

Reply:
https://github.com/microsoft/typespec/pull/3578/files#diff-8942abcaf58f1f649fa1024c634cae80466a5badd90db08b826322a251a952cb